docker/gpu-cuda/jax.dockerfile

```diff
@@ -0,0 +1,46 @@
+ARG BASE_IMAGE=nvidia/cuda:11.1-base
+
+# Setup develop base image packages(build-essentials etc)
+FROM ${BASE_IMAGE} as dev-base
+
+RUN apt-get update && apt-get install -y --no-install-recommends
+RUN DEBIAN_FRONTEND="noninteractive" apt-get install tzdata
+RUN apt-get install -y build-essential \
+        tzdata \
+        ca-certificates \
+        ccache \
+        cmake \
+        curl \
+        git \
+        libjpeg-dev \
+        libpng-dev && \
+    rm -rf /var/lib/apt/lists/*
+RUN /usr/sbin/update-ccache-symlinks
+RUN mkdir /opt/ccache && ccache --set-config=cache_dir=/opt/ccache
+ENV PATH /opt/conda/bin:$PATH
+
+# Setup Miniconda
+FROM dev-base as conda
+ARG PYTHON_VERSION=3.7
+RUN curl -fsSL -v -o ~/miniconda.sh -O  https://repo.anaconda.com/miniconda/Miniconda3-py39_4.9.2-Linux-x86_64.sh  && \
+    chmod +x ~/miniconda.sh && \
```

Comment:
@arshpreetsingh Thanks for the PR! Should we consider adding checksum verification to the miniconda installation script? 🤔 

https://docs.conda.io/en/latest/miniconda.html#linux-installers
https://conda.io/projects/conda/en/latest/user-guide/install/download.html#cryptographic-hash-verification

Reply:
@gtr8 Yes, That option could be added. once it will be finalised either have to keep conda or not(licensing discussion) we can update accordingly.